consensus/polybft/fsm.go

```diff
@@ -429,10 +429,24 @@ func (f *fsm) VerifyStateTransactions(transactions []*types.Transaction) error {
 func (f *fsm) Insert(proposal []byte, committedSeals []*messages.CommittedSeal) (*types.FullBlock, error) {
 	newBlock := f.target
 
+	var proposedBlock types.Block
+	if err := proposedBlock.UnmarshalRLP(proposal); err != nil {
+		return nil, fmt.Errorf("failed to insert proposal, block unmarshaling failed: %w", err)
+	}
+
+	if newBlock == nil || newBlock.Block.Hash() != proposedBlock.Hash() {
+		// if this is the case, we will let syncer insert the block
+		return nil, errors.New(`failed to insert proposal, because the validated proposal is either nil 
+		or it does not match the received one`)
```

Comment:
Nitpick, consistency wise:
```suggestion
		return nil, errors.New("failed to insert proposal, because the validated proposal is either nil 
		or it does not match the received one")
```